src/nbl/ext/ImGui/shaders/common.hlsl

```diff
@@ -9,19 +9,48 @@
 	struct PSInput
 	{
 		float4 position : SV_Position;
-		float4 color    : COLOR0;
 		float2 uv : TEXCOORD0;
+		float4 color    : COLOR0;
+		uint drawID : SV_InstanceID;
+		float clip[4] : SV_ClipDistance;
 	};
 
 	struct PushConstants
 	{
+		uint64_t elementBDA;
+		uint64_t elementCount;
 		float2 scale;
 		float2 translate;
+		float4 viewport;
 	};
+	
 #else
 	struct PushConstants
 	{
+		uint64_t elementBDA;
+		uint64_t elementCount;
 		float scale[2];
 		float translate[2];
+		float viewport[4];
```

Comment:
why are you making a distinction, just use `float32_t2` and `float32_t4` and the types will be available in C++ as long as you `using namespace nbl::hlsl` beforehand

Reply:
> as long as you using namespace nbl::hlsl beforehand

I'm aware, currently I don't have our builtins mounted while compiling, my own extension archive is mounted only - I may add them to see the namespace with our types

Reply:
but to use `float32_tN` in C++ you don't need to mount any builtins, just include the `cpp_compat.hlsl`